https://git.savannah.gnu.org/gitweb/?p=gperf.git;a=commit;h=1a8e476f99335ad5a553f24f1956a084fc6adc10

From 1a8e476f99335ad5a553f24f1956a084fc6adc10 Mon Sep 17 00:00:00 2001
From: Bruno Haible <bruno@clisp.org>
Date: Wed, 9 Apr 2025 13:10:17 +0200
Subject: [PATCH] Fix "make check" failure with an ISO C 23 conforming C
 compiler.

Prompted by
<https://lists.gnu.org/archive/html/bug-gperf/2025-04/msg00002.html>.

* tests/smtp.gperf (my_case_strcmp, main): Convert from K&R C syntax
to ISO C 90 syntax.
---
 ChangeLog        | 8 ++++++++
 tests/smtp.gperf | 8 ++------
 2 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/tests/smtp.gperf b/tests/smtp.gperf
index da6ec7d..a5c448d 100644
--- a/tests/smtp.gperf
+++ b/tests/smtp.gperf
@@ -124,9 +124,7 @@ Xref
 #include <ctype.h>
 
 static int
-my_case_strcmp (s1, s2)
-     register const char *s1;
-     register const char *s2;
+my_case_strcmp (const char *s1, const char *s2)
 {
   for (;;)
     {
@@ -143,9 +141,7 @@ my_case_strcmp (s1, s2)
 }
 
 int
-main (argc, argv)
-     int argc;
-     char *argv[];
+main (int argc, char *argv[])
 {
   int i, j, k, n, exitcode;
   size_t len;
-- 
2.17.1
